torch/_dynamo/external_utils.py

```diff
@@ -97,6 +97,19 @@ def untyped_storage_size(x: torch.Tensor):
     return x.untyped_storage().size()
 
 
+final_callbacks = []
```

Comment:
I worry we will have weird behavior here where if an exception is thrown then `final_callbacks` will be non-empty at the start of the next graph, because the prior graph never called `exec_final_callbacks`.  This will be very hard to debug.  There is also a similar issue with multiple threads.

Can we make this local to the graph?